packages/@aws-cdk/aws-logs-destinations/lib/lambda.ts

```diff
@@ -4,22 +4,54 @@ import * as logs from '@aws-cdk/aws-logs';
 import { Construct } from '@aws-cdk/core';
 
 /**
- * Use a Lamda Function as the destination for a log subscription
+ * Options that may be provided to LambdaDestination
+ */
+export interface LambdaDestinationOptions{
+  /** Whether or not to add Lambda Permissions.
+   * @default true
+   */
+  readonly addPermissions?: boolean | undefined;
+}
+
+const defaultLambdaDestinationOptions: LambdaDestinationOptions = {
```

Comment:
We generally don't do this, but interpret the absence of a value as the advertised default value.

That could take the forms of:

```ts
if (options.addPermissions !== false) {
  // undefined doesn't count as false, which means it gets interpreted the same way as true
}

// or

if (options.addPermissions ?? true) {
  // explicitly switch undefined to true, then compare
}
```